now.json

```diff
@@ -1,4 +1,31 @@
 {
+  "headers": [
+    {
+      "source": "/(.*)",
+      "headers": [
+        {
+          "key": "Access-Control-Allow-Origin",
+          "value": "*"
+        },
+        {
+          "key": "X-Content-Type-Options",
+          "value": "nosniff"
+        },
+        {
+          "key": "Feature-Policy",
+          "value": "*"
+        },
+        {
+          "key": "Referrer-Policy",
+          "value": "origin"
+        },
+        {
+          "key": "Content-Security-Policy",
+          "value": "default-src https: data: 'unsafe-inline'"
```

Comment:
I was helping out [in this Twitter thread](https://twitter.com/excalidraw/status/1246165859397140483) and was told to leave a comment here.

As you probably know, `Content-Security-Policy` is an HTTP security header that lets you lock down what can be done on a page. It's done with a series of directives, like `default-src` or `script-src`. A few examples:

- `Content-Security-Policy: default-src 'none'; script-src foo.example.com` would block everything—CSS, images, websockets—except for JavaScript, which could be loaded from `foo.example.com`.
- `Content-Security-Policy: default-src https:; connect-src bar.example.com` would allow things to be loaded over HTTPS, though various things like WebSockets and AJAX requests would be allowed to talk to `bar.example.com`.

There are [a bunch of different directives that you can read about](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy).

When you add a CSP to your page after it's been around awhile, you may realize that things break in unexpected ways. You set a policy that you _think_ should work, but random things break. In this case, it sounds like some collaboration-related features are broken. There are basically three ways to fix this problem:

1. Give up on CSP altogether (easy, but bad)
2. Try to get your CSP correct the first try, and hope you didn't miss anything (hard)
3. Use the ["report only" version of the header][0] which won't break anything, but it tells you about when things *would have* broken so you can fix them. Eventually, you'll turn on the "real" header.

For existing applications, I tend to recommend option 3. You'll need to add new route to handle these reports. [MDN has some info about this][0] as does [the documentation for a CSP module I maintain][1]. You'll be able to use these reports to make sure your CSP works, tweaking it as you go, and then eventually turn it on for real.

Does that make sense? Does that seem reasonable?

[0]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only
[1]: https://helmetjs.github.io/docs/csp/

Reply:
If we go with option 3. 

do we want to setup react router or is there any way we can add this route?

Reply:
I don't know your setup, but you'd need to create a route that accepts POST requests.

Reply:
Thanks for the detail response @EvanHahn!

I made it work.. and the rules are quite relaxed at the moment.. we can tighten them up as we go!